### PR TITLE
DASD-9562 - Added Function App block and ASP to ARM Template

### PR DESCRIPTION
**Pull request by HussainsGitHub**
DASD-9562 - Added Function App block and App Service Plan block to ARM Template

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -76,8 +111,11 @@
     "variables": {
         "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
         "resourceNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName'),'-', parameters('serviceName')))]",
+        "functionAppNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName'),'-', parameters('functionAppServiceName')))]",
         "resourceGroupName": "[concat(variables('resourceNamePrefix'), 'api-rg')]",
         "appServiceName": "[concat(variables('resourceNamePrefix'), 'api-as')]",
+        "functionAppName": "[concat(variables('functionAppNamePrefix'), '-fa')]",
+        "appServicePlanName": "[concat(variables('resourceNamePrefix'), '-asp')]",
```
Comment (cpwevans) [inline azure/template.json:118]: Might be worth changing resourceNamePrefix to functionAppNamePrefix so it is clear that the app service plan is specific to the function app.

File: pipeline-templates/job/deploy.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ parameters:
   OverrideBlockOnPossibleDataLoss: false
 
 jobs:
-- deployment: DeployWebApp
+- deployment: DeployFunctionApp
```
Comment (cpwevans) [inline pipeline-templates/job/deploy.yml:7]: This shouldn't be FunctionApp as this job deploys the arm, api, jobs and sql database. Maybe just Deploy or DeployApps

**Review by cpwevans — APPROVED** [review]

File: pipeline-templates/job/deploy.yml (hunk 2 above)
Comment (HussainsGitHub) [reply pipeline-templates/job/deploy.yml]: "DeployFunctionApp" has been changed to "DeployApps".

File: azure/template.json (hunk 1 above)
Comment (HussainsGitHub) [reply azure/template.json]: AppServicePlanName has been changed to use the fucntionAppNamePrefix.

**Pull request merged**